onnxruntime/test/framework/execution_frame_test.cc

```diff
@@ -254,6 +254,82 @@ TEST_F(ExecutionFrameTest, MemPatternTest) {
   ASSERT_EQ(p->GetBlock(4)->offset_, kAllocAlignment);
 }
 
+#if defined(ENABLE_TRAINING) || defined(ENABLE_TRAINING_OPS)
+TEST_F(ExecutionFrameTest, MemPatternWithExternalOutputsTest) {
+  auto cpu_xp = CreateCPUExecutionProvider();
```

Comment:
Is it possible to actually execute the model to validate behavior? This test seems to pull out a bunch of internals to manually test the pieces involved, but without a full end to end test via session Run there could be gaps in coverage.